processing/src/main/java/io/druid/query/select/SelectQuery.java

```diff
@@ -65,9 +64,8 @@ public SelectQuery(
       @JsonProperty("context") Map<String, Object> context
   )
   {
-    super(dataSource, querySegmentSpec, descending, context);
+    super(dataSource, querySegmentSpec, descending, context, granularity == null ? Granularities.ALL : granularity);
```

Comment:
I suggest to extract method `Granularities.nullToAll()`

Reply:
:+1: